src/app/pages/wallets/my-wallets/wallet-form-modal/wallet-form-modal.component.ts

```diff
@@ -17,15 +17,38 @@ export class WalletFormModalComponent {
         @Inject(MAT_DIALOG_DATA) public data: MyWallet | undefined,
     ) {
         this.form = new FormGroup<IWalletModalFormType>({
-            name: new FormControl(this.data?.name ?? '', { nonNullable: true }),
+            name: new FormControl(this.data?.name ?? '', {
+              validators: [
+                Validators.required,
+                Validators.maxLength(20),
+              ],
+              nonNullable: true,
+            }),
         });
     }
 
+    public get nameValidator(): boolean {
```

Comment:
If you are using something inside `*ngIf` you can check if the name of it is okay by answering question "if what?". For example in that case it will be "if nameValidator" which doesn't make sense. Better would be for example "if nameIsInvalid". Same issue with max length error.